src/Processors/Formats/Impl/PrettyCompactBlockOutputFormat.cpp

```diff
@@ -57,7 +57,8 @@ PrettyCompactBlockOutputFormat::PrettyCompactBlockOutputFormat(WriteBuffer & out
 void PrettyCompactBlockOutputFormat::writeHeader(
     const Block & block,
     const Widths & max_widths,
-    const Widths & name_widths)
+    const Widths & name_widths,
+    const bool write_footer)
```

Comment:
The footer and header only have a few differences. I wanted to template this function as `writeHeader<is_footer>` and add a `if constexpr` check, but it's simpler to add a new parameter here otherwise I need to make changes in both .cpp and .h files. 

When something needs to change for the footer it is found under `if (write_footer)`

Reply:
Yes, and a regular function argument is better than a template :+1: